### PR TITLE
[feat] support volumesnapshotclass in helm chart

### LINKED ISSUE #1710: [Helm chart] add support for VolumeSnapshotClass
**Is your feature request related to a problem?/Why is this needed**
The helm-chart (GA and 2.0 beta) are currently lacking the support of adding `VolumeSnapshotClass` next to the CSI installation.

It is covered in the [documentation](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/example/snapshot/storageclass-azuredisk-snapshot.yaml) and would ease up things, having this directly in the chart. I have also created a feature request for the [`velero` chart](https://github.com/vmware-tanzu/helm-charts/issues/434), since this is our use case, but from my perspective it’s fair to have it in *both* charts. Here, to support it during the installation of the `azuredisk-csi` which would allow Snapshotting with whatever tool one wants ot use and in the Velero chart for the usage of `velero` when the `csi-driver` does not come with the support for this.

**Describe the solution you'd like in detail**
I would like to have the possibility for:

```
createVolumeSnapshotClass: enabled # defaults to disabled
volumeSnapshotClasses:
  - name: csi-azuredisk-vsc
    driver: disk.csi.azure.com
    deletionPolicy: Delete
    parameters:
      - incremental: true
    additionalLabels:  
      - velero.io/csi-volumesnapshot-class: "true" should always be added 
      - label1: true
      - label2: test
```

**Describe alternatives you've considered**
It’s pretty use to just `kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/master/deploy/example/snapshot/storageclass-azuredisk-snapshot.yaml`, but also pretty hacky.

**Additional context**
Also here: I’m willing to implement this, but I would require a :+1: or to be confident, that this proposed feature request could be merged.

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

support volumesnapshotclass in helm chart

**Which issue(s) this PR fixes**:

Fixes #1710 

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
```
$ k describe volumesnapshotclass csi-azuredisk-vsc
WARNING: version difference between client (1.26) and server (1.24) exceeds the supported minor version skew of +/-1
Name:             csi-azuredisk-vsc
Namespace:        
Labels:           <none>
Annotations:      helm.sh/hook: post-install
API Version:      snapshot.storage.k8s.io/v1
Deletion Policy:  Delete
Driver:           disk.csi.azure.com
Kind:             VolumeSnapshotClass
Metadata:
  Creation Timestamp:  2023-02-15T07:30:35Z
  Generation:          1
  Managed Fields:
    API Version:  snapshot.storage.k8s.io/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:deletionPolicy:
      f:driver:
      f:metadata:
        f:annotations:
          .:
          f:helm.sh/hook:
      f:parameters:
        .:
        f:incremental:
    Manager:         helm
    Operation:       Update
    Time:            2023-02-15T07:30:35Z
  Resource Version:  256378
  UID:               58e5dc75-a961-42fc-821c-304eeb699d8e
Parameters:
  Incremental:  true
Events:         <none>
```
```
$ k get volumesnapshotclass csi-azuredisk-vsc -o yaml
WARNING: version difference between client (1.26) and server (1.24) exceeds the supported minor version skew of +/-1
apiVersion: snapshot.storage.k8s.io/v1
deletionPolicy: Delete
driver: disk.csi.azure.com
kind: VolumeSnapshotClass
metadata:
  annotations:
    helm.sh/hook: post-install
  creationTimestamp: "2023-02-15T07:30:35Z"
  generation: 1
  name: csi-azuredisk-vsc
  resourceVersion: "256378"
  uid: 58e5dc75-a961-42fc-821c-304eeb699d8e
parameters:
  incremental: "true"
```

**Release note**:
```
none
```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

File: charts/latest/azuredisk-csi-driver/values.yaml [hunk 1]
```diff
@@ -145,6 +145,13 @@ snapshot:
       requests:
         cpu: 10m
         memory: 20Mi
+  VolumeSnapshotClass:
+    enabled: false
+    name: csi-azuredisk-vsc
+    deletionPolicy: Delete
+    parameters:
+      incremental: '"true"' # available values: "true", "false" ("true" by default for Azure Public Cloud, and "false" by default for Azure Stack Cloud)
```
Comment (andyzhangx) [inline charts/latest/azuredisk-csi-driver/values.yaml:153]: pls also add `resourceGroup`, `tags` parameters:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/docs/driver-parameters.md#volumesnapshotclass

Comment (umagnus) [reply charts/latest/azuredisk-csi-driver/values.yaml:153]: ok, they are added.

**Review by andyzhangx — COMMENTED** [review]
pls also add `--set xxx=true` in below config, that could make sure e2e test would run with snapshot class creation, thus it would get e2e test coverage.

https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/d9cbdc6f6c9be7529fe53e9a7dcfbb3fa3b909af/Makefile#L43

**Comment by andyzhangx** [comment]
> pls also add `--set xxx=true` in below config, that could make sure e2e test would run with snapshot class creation, thus it would get e2e test coverage.
> 
> https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/d9cbdc6f6c9be7529fe53e9a7dcfbb3fa3b909af/Makefile#L43

nvm, I have made the change, need to merge this change tomorrow, and then cut release v1.27.0 tomorrow, a little tight schedule.

**Comment by umagnus** [comment]
> 

I think this should also set snapshot.enabled=true, because snapshot default is false.

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**